_includes/tutorials/message-ordering/kafka/code/src/main/java/io/confluent/developer/KafkaProducerApplication.java

```diff
@@ -0,0 +1,111 @@
+package io.confluent.developer;
+
+
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.serialization.StringSerializer;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+
+public class KafkaProducerApplication {
+
+    private final Producer<String, String> producer;
+    final String outTopic;
+
+    public KafkaProducerApplication(final Producer<String, String> producer,
+                                    final String topic) {
+        this.producer = producer;
+        outTopic = topic;
+    }
+
+    public Future<RecordMetadata> produce(final String message) {
+        final String[] parts = message.split("-");
+        final String key, value;
+        if (parts.length > 1) {
+            key = parts[0];
+            value = parts[1];
+        } else {
+            key = "NO-KEY";
+            value = parts[0];
+        }
+        final ProducerRecord<String, String> producerRecord = new ProducerRecord<>(outTopic, key, value);
+        return producer.send(producerRecord);
```

Comment:
It's probably more common to provide a [callback](https://www.javadoc.io/static/org.apache.kafka/kafka-clients/2.6.0/org/apache/kafka/clients/producer/KafkaProducer.html#send-org.apache.kafka.clients.producer.ProducerRecord-org.apache.kafka.clients.producer.Callback-) to asynchronously handle acks, instead of handling the `Future` returned by `send` directly

Reply:
cc: @bbejeck also borrowed from the other KT.  Any leaning on how to fix and where?

Reply:
@bbejeck did you have a leaning on whether to leave this as-is, or whether to implement `Callback`?

Reply:
Thanks for discussing @bbejeck , as we agreed, I will add Callbacks just to this KT and no impact to other KTs

Reply:
@bbejeck @mikebin https://github.com/confluentinc/kafka-tutorials/pull/587/commits/4214a91f008525b00d78f89479dcca72cfd89fde implements the callback